### ISSUE TITLE
Query::Arguments#to_h may return nested Arguments objects

### ISSUE THREAD

**Issue by kliuless** [rmosolgo/graphql-ruby#81]
I have a mutation that accepts an input_field `a`, which is an input object type with a field `b`.
In the mutation resolver, the inputs argument is something like this:
```
#<GraphQL::Query::Arguments:0x007f993d572aa0
 @hash={"a"=>#<GraphQL::Query::Arguments:0x007f993d572960 @hash={"b"=>5}, @values={"b"=>5}>},
 @values={"a"=>#<GraphQL::Query::Arguments:0x007f993d572960 @hash={"b"=>5}, @values={"b"=>5}>}>
```
calling `to_h` on it merely returns `@hash`, so I need to call `to_h` at every depth.

When calling `Arguments.new` on a nested hash, it recursively creates `Arguments` objects when setting `@value`, but the original is retained in `@hash`.
However when handling a real request, even the `@hash` has nested `Arguments` objects.
Is there some built-in way to unpack the `Arguments` recursively?

**Comment by davidnorth** [comment]
When I added that feature I just wanted to turn the arguments for my object input types into a hash for Rails' sake, I hadn't considered converting the entire args object. I agree that'd be the expected behaviour. It'd be easy enough to implement if you'd like to have a go.

**Comment by rmosolgo** [comment]
No built-in way, but +1 for recursive `to_h`-ing!

**Comment by rmosolgo** [comment]
fixed by 6e3d6f0
